app/forms/common/Date.scala

```diff
@@ -55,58 +55,50 @@ object Date {
   val twoDigitFormatter = new DecimalFormat("00")
   val fourDigitFormatter = new DecimalFormat("0000")
 
-  def validate(day: Try[Int], month: Try[Int], year: Try[Int]): Boolean = build(day, month, year).isSuccess
-  def validateNonEmptyFields(day: String, month: String, year: String): Boolean = Seq(day, month, year).forall(_.nonEmpty)
-  def isAnyFieldNotEmpty(fields: Seq[String]): Condition = mapping => fields.exists(field => mapping.getOrElse(field, "").nonEmpty)
+  def isValidDate(day: String, month: String, year: String): Boolean =
+    if (isAnyFieldEmpty(Seq(day, month, year))) true
+    else Try(LocalDate.of(year.toInt, month.toInt, day.toInt)).isSuccess
+  def isAnyFieldNotEmpty(fields: Seq[String]): Boolean = fields.exists(_.nonEmpty)
```

Comment:
Maybe rename to `isAnyFieldPopulated` (save the sort of double negative use of 'not' & 'empty' together)?